### ISSUE TITLE
Warning message on PHP files consisting of only comments

### ISSUE THREAD

**Issue by fitztrev** [nikic/PHP-Parser#589]
I'm using [`rectorphp/rector`](https://github.com/rectorphp/rector) and when I run the following command, I get a warning message when it found a PHP file where the contents were only comments.

```
$ ./vendor/bin/rector process app --level laravel58
```

Contents of *`app/Test.php`*:

```php
<?php

// A single-line comment
```

Output:

```
PHP Warning:  assert(): assert($itemStartPos >= 0 && $itemEndPos >= 0) failed in
vendor/nikic/php-parser/lib/PhpParser/PrettyPrinterAbstract.php on line 753
```

**Comment by fitztrev** [comment]
It seems the issue is only on PHP 7.2. Just tried on 7.3 and the warning is not there.

**Comment by fitztrev** [comment]
Had to do with PHP `zend.assertions` config. See rectorphp/rector#1157 for more details.

**Comment by TomasVotruba** [comment]
It still should pass even with assertions on. I'll try to investigate, what is causing this.

Probably some attribute copying

**Comment by fitztrev** [comment]
Re-opening because I was able to generate some steps to reproduce and a failing test here: https://github.com/rectorphp/rector/pull/1284

I'll leave it to this project's maintainers to determine how/if they would like to handle. Thanks

**Comment by nikic** [comment]
Can repro. I think that the end attributes on the nop statement are completely missing in this case, which is why we get a -1 token end pos.

**Comment by fitztrev** [comment]
I just attempted a fix in #601. This was my first dive into this code so it could definitely use a review. Could either of you take a look and let me know? @nikic @TomasVotruba
